### ISSUE TITLE
MDBOOK002: False positives on external doc links (../std/, ../reference/)

### ISSUE THREAD

**Issue by joshrotenberg** [joshrotenberg/mdbook-lint#337]
## Description

When linting The Rust Book, MDBOOK002 flags links like `../std/index.html` and `../reference/items/unions.html` as broken internal links.

These are **not broken** - they're links to external documentation (rustdoc, reference) that only resolve at runtime when the book is hosted alongside the Rust docs on doc.rust-lang.org.

## Examples

```
error[MDBOOK002]: Internal link '../std/index.html' does not resolve to an existing file
error[MDBOOK002]: Internal link '../reference/items/unions.html' does not resolve to an existing file
error[MDBOOK002]: Internal link '../nomicon/vec/vec.html' does not resolve to an existing file
```

## Suggested Fix

Options:
1. Add configuration to specify known external path prefixes to ignore (e.g., `../std/`, `../reference/`, `../nomicon/`)
2. Only validate links that start with `./` or are relative within the book structure
3. Add a heuristic: if a `../` link points outside the book root, skip validation

## Found While Testing

Ran mdbook-lint against rust-lang/book - found 34 false positives for this rule.
